pycm/generate_random_data.py

```diff
@@ -0,0 +1,143 @@
+    elif scenario == Scenario.MINORITY_CLASS:
+        # One class has a minority percentage, others share the rest equally
+        minority_percentage = (100 / num_classes) * 0.2
+        remaining_percentage = (100 - minority_percentage) / (num_classes - 1)
+        return [minority_percentage] + [remaining_percentage] * (num_classes - 1)
+    else:
+        raise ValueError("Invalid scenario")
+
+
+def _calculate_class_counts(class_percentages, total_population):
+    """
+    Calculate the number of samples for each class based on percentages and total population.
+
+    :param class_percentages: dict of percentages for each class (sum should be 100)
+    :type class_percentages: dict
+    :param total_population: total number of samples
+    :type total_population: int
+    :return: dictionary of sample counts for each class
+    """
+    classes = list(class_percentages.keys())
+    percentages = np.array(list(class_percentages.values()), dtype=float)
+
+    if len(classes) < 2:
+        raise ValueError("Number of classes must be at least 2.")
+
+    normalized_percentages = percentages / percentages.sum()
+    class_counts = (normalized_percentages * total_population).astype(int)
+
+    # Handle rounding errors
+    remainder = total_population - class_counts.sum()
+    if remainder > 0:
+        class_counts[np.argmax(normalized_percentages)] += remainder
+
+    return dict(zip(classes, class_counts.astype(int).tolist()))
+
+
+def generate_confusion_matrix(class_percentages, total_population):
+    """
+    Generate a random confusion matrix with given class percentages and total population.
+
+    :param class_percentages: dict or list of percentages for each class (sum should be 100)
+    :type class_percentages: dict | list
+    :param  total_population: total number of samples in the confusion matrix
+    :type total_population: int
+    :return: confusion matrix as a dictionary
+    """
+    if total_population <= 0:
+        raise ValueError("Total population must be positive.")
+    if isinstance(class_percentages, list):
+        class_percentages = dict(enumerate(class_percentages))
+
+    class_labels = list(class_percentages.keys())
+    num_classes = len(class_percentages)
+
+    if num_classes < 2:
+        raise ValueError("Number of classes must be at least 2.")
+
+    class_counts = _calculate_class_counts(class_percentages, total_population)
+    confusion_matrix = {
+        actual: {pred: 0 for pred in class_labels} for actual in class_labels
+    }
+
+    for actual in class_labels:
+        count = class_counts[actual]
+        if count == 0:
+            continue
+
+        dirichlet_params = np.ones(num_classes)
+        actual_idx = class_labels.index(actual)
+        dirichlet_params[actual_idx] *= 10  # Bias toward correct class
+
+        probs = np.random.dirichlet(dirichlet_params)
+        predicted_counts = (probs * count).astype(int)
+
+        # Handle rounding remainder
```

Comment:
remove this comment.
Also I think this function can be refactored in a post-facto PR.